test/ami-existence-poll_test.js

```diff
@@ -0,0 +1,254 @@
+const {AmiExistencePoller} = require('../lib/ami-existence-poller');
+const sinon = require('sinon');
+const main = require('../lib/main');
+const assume = require('assume');
+
+describe('AMI Existence Poller', () => {
+  let state;
+  let sandbox = sinon.sandbox.create();
+  let defaultSR;
+
+  before(async() => {
+    state = await main('state', {profile: 'test', process: 'test'});
+    db = await main('state', {profile: 'test', process: 'test'});
```

Comment:
no need to create a second copy of the state.  let's pick either `db` or `state` and drop the other.